charts/jenkins/jobs/nuxeo/lts/nuxeo-es-7.17.groovy.gotmpl

```diff
@@ -0,0 +1,52 @@
+pipelineJob('nuxeo/lts/nuxeo-es-7.17') {
+  // Description
+  description('Test Nuxeo against Elasticsearch 7.17')
+  definition {
+    // Definition
+    cpsScm {
+      scm {
+        // SCM
+        git {
+          // Repositories
+          remote {
+            // Repository URL
+            url('https://github.com/nuxeo/nuxeo-lts')
+            // Credentials
+            credentials('github-branch-source')
+          }
+          // Branches to build
+          branches('^${NUXEO_BRANCH}')
+          browser {
+            // Repository browser
+            githubWeb {
+              // URL
+              repoUrl('https://github.com/nuxeo/nuxeo-lts')
+            }
+          }
+        }
+      }
+      // no lightweight checkout in order to make branch parameter work
+      lightweight(false)
+      // Script Path
+      scriptPath('ci/Jenkinsfiles/build-compat-es-7.17.groovy')
+    }
+  }
+  parameters {
+    string {
+      name('NUXEO_BRANCH')
+      description('Branch to test.')
+      trim(true)
+    }
+    string {
+      name('NUXEO_BUILD_VERSION')
+      defaultValue('')
+      description('Version of the Nuxeo Server build to test.')
+      trim(true)
+    }
+  }
```

Comment:
I'm wondering why we need both parameters? Just one seems enough me.

Reply:
We take the pipeline from the `NUXEO_BRANCH` repo but we test it on a tag `NUXEO_BUILD_VERSION`
This allows testing further pipeline modifications without merging them.
If I understand correctly this is what is done also on other pipelines.
Here in my test
`NUXEO_BRANCH` = `tast-NXP-31784-test-against-elasticsearch-7.17.9`
`NUXEO_BUILD_VERSION`= `2021.36.4`

Reply:
But wouldn't I have to do this everytime I do a pipeline modification PR ?

Reply:
@kevinleturc indeed if we want to validate ES 7.17 for a given branch then using `git checkout v${...}` won't work (https://github.com/nuxeo/nuxeo-lts/pull/1082/files#diff-27815d086a6be2c3e2f3ca82ce2b9866fa432d6856f5a03a497f8b942dfd7d82R60), we would have to pass the tag with v in the parameter.
That's if we stick with 2 params.

If you think 2 params are confusing I'm OK to switch to one.
My main argument was to be consistent with the other release and post-release pipelines, `NUXEO_BRANCH` often referring to 2021 or 2023.

Reply:
One thing remains. If we give a real branch names, it will likely hit the 63 chars limitation unless they are main branches or tags with a short name.

Reply:
Then you could truncate the branch name for namespace generation.

Reply:
what about keeping only the ticket ID for example ?
`NXP-12345`

Reply:
sure, if the logic is not too complex (think of edge cases, branches without ticket, etc.), otherwise truncate to be below 63 including the prefix/suffix seams good to me

Reply:
Ok for me. This is certainly safer and easier.

Reply:
>If you think 2 params are confusing I'm OK to switch to one.
My main argument was to be consistent with the other release and post-release pipelines, NUXEO_BRANCH often referring to 2021 or 2023.

I agree that the parameter is not well named, the benchmark job has this naming too for the same case. I named the parameter like this because it has this name in the Jenkins DSL. Maybe something like `NUXEO_GIT_REF` could do the job?

Reply:
My main argument to have only one parameter is indeed to be able to test a fix not yet merged on ES 7.17. But indeed we're losing consistency with other jobs and the capability to run a different pipeline on a specific nuxeo version.

Maybe we can talk about it in a meeting?

Reply:
Recap:
We agreed on `NUXEO_BRANCH` as unique job parameter.
Depending on the fact that the branch is a tag or not, artifacts will be pulled from the nexus or built locally.